pkg/core/native/native_test/cryptolib_verification_test.go

```diff
@@ -0,0 +1,924 @@
+		sigs := make([][]byte, m)
+		for i := range sigs {
+			j := i
+			if i > 0 {
+				j++ // Add some shift to ensure that verification script works correctly.
+			}
+			if i > 3 {
+				j++ // Add more shift for large number of public keys for the same purpose.
+			}
+			sigs[i] = pks[j].SignHash(hash.Keccak256(msg))
+		}
+
+		// Build invocation witness script for the signatures.
+		invBytes := buildKoblitzInvocationScript(t, sigs)
+
+		// Construct witness for signer #0 (the multisig account itself).
+		tx.Scripts = []transaction.Witness{
+			{
+				InvocationScript:   invBytes,
+				VerificationScript: vrfBytes,
+			},
+		}
+
+		// Add transaction to the chain. No error is expected on new block addition. Note, that this line performs
+		// all those checks that are executed during transaction acceptance in the real network.
+		e.AddNewBlock(t, tx)
+
+		// Double-check: ensure funds have been transferred.
+		e.CheckGASBalance(t, to, big.NewInt(int64(amount)))
+	}
+
+	// The proposed multisig verification script.
+	// (279 bytes, 8387820 GAS including Invocation script execution for 3/4 multisig).
+	// The user has to sign the keccak256([4-bytes-network-magic-LE, txHash-bytes-BE]).
+	check(t, buildKoblitzMultisigVerificationScript, constructMessage)
+}
+
+// buildKoblitzMultisigVerificationScript builds witness verification script for m signatures out of n Koblitz public keys.
+// Public keys must be sorted. Signatures (pushed by witness Invocation script) must be sorted by public keys.
+// It checks m out of n multisignature of the following message:
+//
+//	keccak256([4-bytes-network-magic-LE, txHash-bytes-BE])
+func buildKoblitzMultisigVerificationScript(t *testing.T, m int, pubs keys.PublicKeys) []byte {
+	if m > len(pubs) {
+		t.Fatalf("m must be not greater than the number of public keys")
+	}
+
+	criptoLibH := state.CreateNativeContractHash(nativenames.CryptoLib)
+	sort.Sort(pubs)
+
+	// In fact, the following algorithm is implemented via NeoVM instructions:
+	//
+	// func Check(sigs []interop.Signature) bool {
+	// 	var m int = 3
+	// 	if m != len(sigs) {
+	// 		return false
+	// 	}
+	// 	var pubs []interop.PublicKey = []interop.PublicKey{}
+	// 	msg := append(convert.ToBytes(runtime.GetNetwork()), runtime.GetScriptContainer().Hash...)
+	// 	var numOK int
+	// 	var pubsStart = 0
+	// 	for i := 0; i < m; i++ { // sigs must be sorted by pub
+	// 		for j := pubsStart; j < len(pubs); j++ { // pubs must be sorted
+	// 			pubOK := crypto.VerifyWithECDsa(msg, pubs[j], sigs[i], crypto.Secp256r1)
+	// 			if pubOK {
+	// 				numOK++
+	// 				pubsStart = j + 1
+	// 				break
+	// 			}
+	// 		}
+	// 	}
+	// 	return numOK == m
+	// }
+	vrf := io.NewBufBinWriter()
+
+	// Initialize slots for local variables. Locals slot scheme:
+	// LOC0 -> sigs
+	// LOC1 -> pubs
+	// LOC2 -> msg (ByteString)
+	// LOC3 -> numOk (Integer)
+	// LOC4 -> pubsStart (Integer)
+	// LOC5 -> i (Integer) - signatures range index
+	// LOC6 -> j (Integer) - pubs range index
```

Comment:
I think you can merge `j` and `pubsStart`. This will simplify things a bit and make script closer to spending a constant GAS amount.